.github/workflows/_pr_entrypoint.yaml

```diff
@@ -0,0 +1,242 @@
+name: PR Entrypoint
+
+concurrency:
+  group: pr-entrypoint-${{ github.event_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      ref:
+        required: false
+
+env:
+  IS_CI: "yes"
+
+jobs:
+  sanity-checks:
+    runs-on: ${{ github.repository_owner == 'emqx' && 'aws-amd64' || 'ubuntu-22.04' }}
+    container: "ghcr.io/emqx/emqx-builder/5.1-3:1.14.5-25.3.2-1-ubuntu22.04"
+    outputs:
+      ct-matrix: ${{ steps.matrix.outputs.ct-matrix }}
+      ct-host: ${{ steps.matrix.outputs.ct-host }}
+      ct-docker: ${{ steps.matrix.outputs.ct-docker }}
+      version-emqx: ${{ steps.matrix.outputs.version-emqx }}
+      version-emqx-enterprise: ${{ steps.matrix.outputs.version-emqx-enterprise }}
+      runner: ${{ github.repository_owner == 'emqx' && 'aws-amd64' || 'ubuntu-22.04' }}
+      builder: "ghcr.io/emqx/emqx-builder/5.1-3:1.14.5-25.3.2-1-ubuntu22.04"
+      builder_vsn: "5.1-3"
+      otp_vsn: "25.3.2-1"
+      elixir_vsn: "1.14.5"
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.ref }}
+          fetch-depth: 0
+      - name: Work around https://github.com/actions/checkout/issues/766
+        run: |
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+      - name: Run gitlint
+        env:
+          BEFORE_REF: ${{ github.event_name == 'pull_request' && github.event.pull_request.base.sha || github.event.before }}
+          AFTER_REF: ${{ github.sha }}
+        run: |
+          pip install gitlint
+          gitlint --commits $BEFORE_REF..$AFTER_REF --config .github/workflows/.gitlint
+      - name: Run shellcheck
+        run: |
+          DEBIAN_FRONTEND=noninteractive apt-get update -qy && apt-get install -qy shellcheck
+          ./scripts/shellcheck.sh
+      - name: Run shell tests
+        run: |
+          DEBIAN_FRONTEND=noninteractive apt-get update -qy && apt-get install -qy shelltestrunner
+          scripts/shelltest/run_tests.sh
+      - name: Check workflow files
+        env:
+          ACTIONLINT_VSN: 1.6.25
+        run: |
+          wget https://github.com/rhysd/actionlint/releases/download/v${ACTIONLINT_VSN}/actionlint_${ACTIONLINT_VSN}_linux_amd64.tar.gz
+          tar zxf actionlint_${ACTIONLINT_VSN}_linux_amd64.tar.gz actionlint
+          # TODO: enable shellcheck when all the current issues are fixed
```

Comment:
Q: we already have shellcheck a couple steps above.  Is this something else?

Reply:
this one would run shellcheck on the content of `run` steps in workflows.